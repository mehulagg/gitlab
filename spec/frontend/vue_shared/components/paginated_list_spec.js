import Vue from 'vue';
import { mount, shallowMount } from '@vue/test-utils';
import PaginatedList from '~/vue_shared/components/paginated_list.vue';

describe('Resizable Chart Container', () => {
  let template;

  beforeAll(() => {
    template = `
      <div class="slot" slot-scope="{ listItem }">
        <span class="item">Item Name: {{listItem.id}}</span>
      </div>
    `;
  });

  describe('Searchless states', () => {
    let wrapper;

    afterEach(() => {
      wrapper.destroy();
    });

    it('renders the list in an empty state, when no items are provided', () => {
      wrapper = shallowMount(PaginatedList, {
        scopedSlots: {
          default: template,
        },
        propsData: {
          list: [],
        },
      });
      expect(wrapper.findAll('.item').length).toEqual(0);
      expect(wrapper.findAll('.empty-message').length).toEqual(1);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders the list in an empty state with a custom empty message, when one is provided', () => {
      const msg = 'List is passed in empty';

      wrapper = shallowMount(PaginatedList, {
        scopedSlots: {
          default: template,
        },
        propsData: {
          list: [],
          emptyMessage: msg,
        },
      });
      expect(wrapper.findAll('.item').length).toEqual(0);
      expect(wrapper.find('.empty-message').text()).toEqual(msg);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders the list with nested items, when an array of items is provided', () => {
      wrapper = shallowMount(PaginatedList, {
        propsData: {
          list: [{ id: 'foo' }, { id: 'bar' }],
        },
        scopedSlots: {
          default: template,
        },
      });
      expect(wrapper.findAll('.item').length).toEqual(2);
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('Search states', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallowMount(PaginatedList, {
        propsData: {
          list: [{ id: 'foo' }, { id: 'bar' }, { id: 'baz' }],
        },
        scopedSlots: {
          default: template,
        },
      });
    });

    afterEach(() => {
      wrapper.destroy();
    });

    it('renders the list flitered by search results', done => {
      wrapper.vm.queryStr = 'ba';

      return Vue.nextTick()
        .then(() => {
          expect(wrapper.findAll('.item').length).toEqual(2);
          expect(wrapper.element).toMatchSnapshot();
          done();
        })
        .catch(done.fail);
    });

    it('renders the default empty search message when there are no search results', done => {
      wrapper.vm.queryStr = 'qux';

      return Vue.nextTick()
        .then(() => {
          expect(wrapper.findAll('.item').length).toEqual(0);
          expect(wrapper.findAll('.empty-search').length).toEqual(1);
          expect(wrapper.element).toMatchSnapshot();
          done();
        })
        .catch(done.fail);
    });

    it('renders the custom empty search message when there are no search results', done => {
      const msg = 'Custom empty message';

      wrapper = shallowMount(PaginatedList, {
        propsData: {
          list: [{ id: 'foo' }, { id: 'bar' }, { id: 'baz' }],
          emptySearchMessage: msg,
        },
        scopedSlots: {
          default: template,
        },
      });

      wrapper.vm.queryStr = 'qux';

      return Vue.nextTick()
        .then(() => {
          expect(wrapper.findAll('.item').length).toEqual(0);
          expect(wrapper.find('.empty-search').text()).toEqual(msg);
          expect(wrapper.element).toMatchSnapshot();
          done();
        })
        .catch(done.fail);
    });
  });

  describe('Pagination', () => {
    let wrapper;

    const list = [
      { id: 'foo' },
      { id: 'bar' },
      { id: 'baz' },
      { id: 'qux' },
      { id: 'quux' },
      { id: 'corge' },
      { id: 'grault' },
      { id: 'garply' },
      { id: 'waldo' },
      { id: 'fred' },
      { id: 'xyzzy' },
      { id: 'plugh' },
      { id: 'thud' },
    ];

    afterEach(() => {
      wrapper.destroy();
    });

    it('renders 10 items for a default page size of 10, 13 total and renders pagination', () => {
      wrapper = mount(PaginatedList, {
        propsData: {
          list,
        },
        scopedSlots: {
          default: template,
        },
      });

      expect(wrapper.findAll('.item').length).toEqual(10);
      expect(wrapper.findAll('.gl-pagination').length).toEqual(1);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders 13 items for a default page size of 20, 13 total items, and does not render pagination', () => {
      wrapper = mount(PaginatedList, {
        propsData: {
          list,
          perPage: 20,
        },
        scopedSlots: {
          default: template,
        },
      });

      expect(wrapper.findAll('.item').length).toEqual(13);
      expect(wrapper.findAll('.gl-pagination').length).toEqual(0);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders 5 items on page 1 for a page size of 5, 13 total itmes and renders pagination', () => {
      wrapper = mount(PaginatedList, {
        propsData: {
          list,
          perPage: 5,
        },
        scopedSlots: {
          default: template,
        },
      });

      expect(wrapper.findAll('.item').length).toEqual(5);
      expect(wrapper.findAll('.gl-pagination').length).toEqual(1);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders 3 items on page 2 with default page size of 10 and 13 total items', () => {
      wrapper = shallowMount(PaginatedList, {
        propsData: {
          list,
          page: 2,
        },
        scopedSlots: {
          default: template,
        },
      });

      expect(wrapper.findAll('.item').length).toEqual(3);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders 1 item on page 7 with page size of 2 and 13 total items', () => {
      wrapper = shallowMount(PaginatedList, {
        propsData: {
          list,
          page: 7,
          perPage: 2,
        },
        scopedSlots: {
          default: template,
        },
      });

      expect(wrapper.findAll('.item').length).toEqual(1);
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('props', () => {
    let wrapper;

    afterEach(() => {
      wrapper.destroy();
    });

    describe('filterable', () => {
      it('has search enabled by default', () => {
        wrapper = mount(PaginatedList, {
          scopedSlots: {
            default: template,
          },
          propsData: {
            list: [],
          },
        });

        expect(wrapper.findAll('[aria-label="Search"]').length).toEqual(1);
      });

      it('has search disabled when filterable prop set to false', () => {
        wrapper = mount(PaginatedList, {
          scopedSlots: {
            default: template,
          },
          propsData: {
            list: [],
            filterable: false,
          },
        });

        expect(wrapper.findAll('[aria-label="Search"]').length).toEqual(0);
      });
    });

    describe('filterKey', () => {
      it('filters on default "id" key', done => {
        wrapper = shallowMount(PaginatedList, {
          scopedSlots: {
            default: template,
          },
          propsData: {
            list: [{ id: 'foo' }, { id: 'bar' }],
          },
        });

        wrapper.vm.queryStr = 'ba';

        return Vue.nextTick()
          .then(() => {
            expect(wrapper.findAll('.item').length).toEqual(1);
            expect(wrapper.element).toMatchSnapshot();
            done();
          })
          .catch(done.fail);
      });

      it('filters on provided "myKey" key', done => {
        wrapper = shallowMount(PaginatedList, {
          scopedSlots: {
            default: template,
          },
          propsData: {
            list: [{ myKey: 'foo' }, { myKey: 'bar' }],
            filterKey: 'myKey',
          },
        });

        wrapper.vm.queryStr = 'ba';

        return Vue.nextTick()
          .then(() => {
            expect(wrapper.findAll('.item').length).toEqual(1);
            expect(wrapper.element).toMatchSnapshot();
            done();
          })
          .catch(done.fail);
      });

      it('gracefully renders all items and disables filtering when provided an invalid filterKey', done => {
        wrapper = shallowMount(PaginatedList, {
          scopedSlots: {
            default: template,
          },
          propsData: {
            list: [{ myKey: 'foo' }, { myKey: 'bar' }],
            filterKey: 'nonMatchingKey',
          },
        });

        wrapper.vm.queryStr = 'ba';

        return Vue.nextTick()
          .then(() => {
            expect(wrapper.findAll('.item').length).toEqual(2);
            expect(wrapper.element).toMatchSnapshot();
            done();
          })
          .catch(done.fail);
      });
    });
  });
});
