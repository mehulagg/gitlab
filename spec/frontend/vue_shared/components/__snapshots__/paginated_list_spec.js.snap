// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resizable Chart Container Pagination renders 1 item on page 7 with page size of 2 and 13 total items 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between"
  >
     
    <glsearchboxbytype-stub
      class="mt-3 mt-sm-0"
      value=""
    />
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  >
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: thud
        </span>
      </div>
    </li>
  </ul>
   
  <paginationlinks-stub
    change="function () { [native code] }"
    class="d-flex justify-content-center prepend-top-default"
    pageinfo="[object Object]"
  />
   
  <!---->
   
  <!---->
</div>
`;

exports[`Resizable Chart Container Pagination renders 3 items on page 2 with default page size of 10 and 13 total items 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between"
  >
     
    <glsearchboxbytype-stub
      class="mt-3 mt-sm-0"
      value=""
    />
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  >
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: xyzzy
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: plugh
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: thud
        </span>
      </div>
    </li>
  </ul>
   
  <paginationlinks-stub
    change="function () { [native code] }"
    class="d-flex justify-content-center prepend-top-default"
    pageinfo="[object Object]"
  />
   
  <!---->
   
  <!---->
</div>
`;

exports[`Resizable Chart Container Pagination renders 5 items on page 1 for a page size of 5, 13 total itmes and renders pagination 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between"
  >
     
    <div
      class="position-relative ms-no-clear mt-3 mt-sm-0"
      data-v-409eac79=""
    >
      <input
        aria-label="Search"
        class="pl-5 form-control"
        data-v-409eac79=""
        id="__BVID__59"
        placeholder="Search"
        type="text"
        value=""
      />
       
      <div
        class="pl-2 pr-2 position-left-0 position-absolute position-top-0 h-100 d-flex align-items-center text-muted"
        data-v-409eac79=""
      >
        <i
          aria-hidden="true"
          class="fa fa-search"
          data-v-409eac79=""
        />
      </div>
       
      <div
        class="pr-2 pl-2 position-right-0 position-absolute position-top-0 h-100 d-flex align-items-center text-muted"
        data-v-409eac79=""
        style="display: none;"
      >
        <span
          class="loading-container text-center"
          data-v-409eac79=""
          style="display: none;"
        >
          <span
            aria-hidden="true"
            aria-label="Loading"
            class="spinner"
          />
        </span>
         
        <i
          aria-hidden="true"
          class="fa fa-times fa-lg cursor-pointer"
          data-original-title="Clear"
          data-v-409eac79=""
          style="display: none;"
          title=""
        />
      </div>
    </div>
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  >
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: foo
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: bar
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: baz
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: qux
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: quux
        </span>
      </div>
    </li>
  </ul>
   
  <ul
    aria-disabled="false"
    aria-label="Pagination"
    class="gl-pagination d-flex justify-content-center prepend-top-default pagination b-pagination pagination-md"
    first-text="« First"
    last-text="Last »"
    next-text="Next"
    prev-text="Prev"
    role="menubar"
  >
    <!---->
    <li
      aria-hidden="true"
      class="page-item disabled"
      role="none presentation"
    >
      <span
        class="page-link"
      >
        Prev
      </span>
    </li>
    <!---->
    <li
      class="page-item active"
      role="none presentation"
    >
      <a
        aria-checked="true"
        aria-label="Goto page 1"
        aria-posinset="1"
        aria-setsize="3"
        class="page-link btn-primary"
        href="#"
        role="menuitemradio"
        tabindex="0"
        target="_self"
      >
        1
      </a>
    </li>
    <li
      class="page-item"
      role="none presentation"
    >
      <a
        aria-checked="false"
        aria-label="Goto page 2"
        aria-posinset="2"
        aria-setsize="3"
        class="page-link"
        href="#"
        role="menuitemradio"
        tabindex="-1"
        target="_self"
      >
        2
      </a>
    </li>
    <li
      class="page-item"
      role="none presentation"
    >
      <a
        aria-checked="false"
        aria-label="Goto page 3"
        aria-posinset="3"
        aria-setsize="3"
        class="page-link"
        href="#"
        role="menuitemradio"
        tabindex="-1"
        target="_self"
      >
        3
      </a>
    </li>
    <!---->
    <li
      class="page-item"
      role="none presentation"
    >
      <a
        aria-label="Goto next page"
        class="page-link"
        href="#"
        role="menuitem"
        tabindex="-1"
        target="_self"
      >
        <span
          aria-hidden="true"
        >
          Next
        </span>
      </a>
    </li>
    <!---->
  </ul>
   
  <!---->
   
  <!---->
</div>
`;

exports[`Resizable Chart Container Pagination renders 10 items for a default page size of 10, 13 total and renders pagination 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between"
  >
     
    <div
      class="position-relative ms-no-clear mt-3 mt-sm-0"
      data-v-409eac79=""
    >
      <input
        aria-label="Search"
        class="pl-5 form-control"
        data-v-409eac79=""
        id="__BVID__40"
        placeholder="Search"
        type="text"
        value=""
      />
       
      <div
        class="pl-2 pr-2 position-left-0 position-absolute position-top-0 h-100 d-flex align-items-center text-muted"
        data-v-409eac79=""
      >
        <i
          aria-hidden="true"
          class="fa fa-search"
          data-v-409eac79=""
        />
      </div>
       
      <div
        class="pr-2 pl-2 position-right-0 position-absolute position-top-0 h-100 d-flex align-items-center text-muted"
        data-v-409eac79=""
        style="display: none;"
      >
        <span
          class="loading-container text-center"
          data-v-409eac79=""
          style="display: none;"
        >
          <span
            aria-hidden="true"
            aria-label="Loading"
            class="spinner"
          />
        </span>
         
        <i
          aria-hidden="true"
          class="fa fa-times fa-lg cursor-pointer"
          data-original-title="Clear"
          data-v-409eac79=""
          style="display: none;"
          title=""
        />
      </div>
    </div>
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  >
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: foo
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: bar
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: baz
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: qux
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: quux
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: corge
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: grault
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: garply
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: waldo
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: fred
        </span>
      </div>
    </li>
  </ul>
   
  <ul
    aria-disabled="false"
    aria-label="Pagination"
    class="gl-pagination d-flex justify-content-center prepend-top-default pagination b-pagination pagination-md"
    first-text="« First"
    last-text="Last »"
    next-text="Next"
    prev-text="Prev"
    role="menubar"
  >
    <!---->
    <li
      aria-hidden="true"
      class="page-item disabled"
      role="none presentation"
    >
      <span
        class="page-link"
      >
        Prev
      </span>
    </li>
    <!---->
    <li
      class="page-item active"
      role="none presentation"
    >
      <a
        aria-checked="true"
        aria-label="Goto page 1"
        aria-posinset="1"
        aria-setsize="2"
        class="page-link btn-primary"
        href="#"
        role="menuitemradio"
        tabindex="0"
        target="_self"
      >
        1
      </a>
    </li>
    <li
      class="page-item"
      role="none presentation"
    >
      <a
        aria-checked="false"
        aria-label="Goto page 2"
        aria-posinset="2"
        aria-setsize="2"
        class="page-link"
        href="#"
        role="menuitemradio"
        tabindex="-1"
        target="_self"
      >
        2
      </a>
    </li>
    <!---->
    <li
      class="page-item"
      role="none presentation"
    >
      <a
        aria-label="Goto next page"
        class="page-link"
        href="#"
        role="menuitem"
        tabindex="-1"
        target="_self"
      >
        <span
          aria-hidden="true"
        >
          Next
        </span>
      </a>
    </li>
    <!---->
  </ul>
   
  <!---->
   
  <!---->
</div>
`;

exports[`Resizable Chart Container Pagination renders 13 items for a default page size of 20, 13 total items, and does not render pagination 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between"
  >
     
    <div
      class="position-relative ms-no-clear mt-3 mt-sm-0"
      data-v-409eac79=""
    >
      <input
        aria-label="Search"
        class="pl-5 form-control"
        data-v-409eac79=""
        id="__BVID__50"
        placeholder="Search"
        type="text"
        value=""
      />
       
      <div
        class="pl-2 pr-2 position-left-0 position-absolute position-top-0 h-100 d-flex align-items-center text-muted"
        data-v-409eac79=""
      >
        <i
          aria-hidden="true"
          class="fa fa-search"
          data-v-409eac79=""
        />
      </div>
       
      <div
        class="pr-2 pl-2 position-right-0 position-absolute position-top-0 h-100 d-flex align-items-center text-muted"
        data-v-409eac79=""
        style="display: none;"
      >
        <span
          class="loading-container text-center"
          data-v-409eac79=""
          style="display: none;"
        >
          <span
            aria-hidden="true"
            aria-label="Loading"
            class="spinner"
          />
        </span>
         
        <i
          aria-hidden="true"
          class="fa fa-times fa-lg cursor-pointer"
          data-original-title="Clear"
          data-v-409eac79=""
          style="display: none;"
          title=""
        />
      </div>
    </div>
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  >
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: foo
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: bar
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: baz
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: qux
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: quux
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: corge
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: grault
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: garply
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: waldo
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: fred
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: xyzzy
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: plugh
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: thud
        </span>
      </div>
    </li>
  </ul>
   
  <!---->
   
  <!---->
   
  <!---->
</div>
`;

exports[`Resizable Chart Container Search states renders the custom empty search message when there are no search results 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between"
  >
     
    <glsearchboxbytype-stub
      class="mt-3 mt-sm-0"
      value=""
    />
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  />
   
  <paginationlinks-stub
    change="function () { [native code] }"
    class="d-flex justify-content-center prepend-top-default"
    pageinfo="[object Object]"
  />
   
  <!---->
   
  <div
    class="bs-callout bs-callout-warning mt-3 empty-search"
  >
    
    Custom empty message
  
  </div>
</div>
`;

exports[`Resizable Chart Container Search states renders the default empty search message when there are no search results 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between"
  >
     
    <glsearchboxbytype-stub
      class="mt-3 mt-sm-0"
      value=""
    />
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  />
   
  <paginationlinks-stub
    change="function () { [native code] }"
    class="d-flex justify-content-center prepend-top-default"
    pageinfo="[object Object]"
  />
   
  <!---->
   
  <div
    class="bs-callout bs-callout-warning mt-3 empty-search"
  >
    
    Sorry, your filter produced no results.
  
  </div>
</div>
`;

exports[`Resizable Chart Container Search states renders the list flitered by search results 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between"
  >
     
    <glsearchboxbytype-stub
      class="mt-3 mt-sm-0"
      value=""
    />
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  >
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: bar
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: baz
        </span>
      </div>
    </li>
  </ul>
   
  <paginationlinks-stub
    change="function () { [native code] }"
    class="d-flex justify-content-center prepend-top-default"
    pageinfo="[object Object]"
  />
   
  <!---->
   
  <!---->
</div>
`;

exports[`Resizable Chart Container Searchless states renders the list in an empty state with a custom empty message, when one is provided 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between"
  >
     
    <glsearchboxbytype-stub
      class="mt-3 mt-sm-0"
      value=""
    />
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  />
   
  <paginationlinks-stub
    change="function () { [native code] }"
    class="d-flex justify-content-center prepend-top-default"
    pageinfo="[object Object]"
  />
   
  <div
    class="bs-callout bs-callout-warning mt-3 empty-message"
  >
    
    List is passed in empty
  
  </div>
   
  <!---->
</div>
`;

exports[`Resizable Chart Container Searchless states renders the list in an empty state, when no items are provided 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between"
  >
     
    <glsearchboxbytype-stub
      class="mt-3 mt-sm-0"
      value=""
    />
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  />
   
  <paginationlinks-stub
    change="function () { [native code] }"
    class="d-flex justify-content-center prepend-top-default"
    pageinfo="[object Object]"
  />
   
  <div
    class="bs-callout bs-callout-warning mt-3 empty-message"
  >
    
    There are currently no items in this list.
  
  </div>
   
  <!---->
</div>
`;

exports[`Resizable Chart Container Searchless states renders the list with nested items, when an array of items is provided 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between"
  >
     
    <glsearchboxbytype-stub
      class="mt-3 mt-sm-0"
      value=""
    />
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  >
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: foo
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: bar
        </span>
      </div>
    </li>
  </ul>
   
  <paginationlinks-stub
    change="function () { [native code] }"
    class="d-flex justify-content-center prepend-top-default"
    pageinfo="[object Object]"
  />
   
  <!---->
   
  <!---->
</div>
`;

exports[`Resizable Chart Container props filterKey filters on default "id" key 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between"
  >
     
    <glsearchboxbytype-stub
      class="mt-3 mt-sm-0"
      value=""
    />
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  >
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: bar
        </span>
      </div>
    </li>
  </ul>
   
  <paginationlinks-stub
    change="function () { [native code] }"
    class="d-flex justify-content-center prepend-top-default"
    pageinfo="[object Object]"
  />
   
  <!---->
   
  <!---->
</div>
`;

exports[`Resizable Chart Container props filterKey filters on provided "myKey" key 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between"
  >
     
    <glsearchboxbytype-stub
      class="mt-3 mt-sm-0"
      value=""
    />
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  >
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: 
        </span>
      </div>
    </li>
  </ul>
   
  <paginationlinks-stub
    change="function () { [native code] }"
    class="d-flex justify-content-center prepend-top-default"
    pageinfo="[object Object]"
  />
   
  <!---->
   
  <!---->
</div>
`;

exports[`Resizable Chart Container props filterKey gracefully renders all items and disables filtering when provided an invalid filterKey 1`] = `
<div>
  <div
    class="row-content-block second-block d-sm-flex justify-content-between"
  >
     
    <!---->
  </div>
    
  <ul
    class="list-group list-group-flush list-unstyled"
  >
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: 
        </span>
      </div>
    </li>
    <li>
      <div
        class="slot"
      >
        <span
          class="item"
        >
          Item Name: 
        </span>
      </div>
    </li>
  </ul>
   
  <paginationlinks-stub
    change="function () { [native code] }"
    class="d-flex justify-content-center prepend-top-default"
    pageinfo="[object Object]"
  />
   
  <!---->
   
  <!---->
</div>
`;
